controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -103,25 +106,27 @@
 
     private final AtomicReference<EventStreamWriter<ControllerEvent>> requestEventWriterRef = new AtomicReference<>();
     private final AuthHelper authHelper;
+    private final RequestTracker requestTracker;
 
     public StreamMetadataTasks(final StreamMetadataStore streamMetadataStore,
                                final HostControllerStore hostControllerStore, final TaskMetadataStore taskMetadataStore,
                                final SegmentHelper segmentHelper, final ScheduledExecutorService executor, final String hostId,
-                               final ConnectionFactory connectionFactory, AuthHelper authHelper) {
+                               final ConnectionFactory connectionFactory, AuthHelper authHelper, RequestTracker requestTracker) {
         this(streamMetadataStore, hostControllerStore, taskMetadataStore, segmentHelper, executor, new Context(hostId),
-                connectionFactory, authHelper);
+                connectionFactory, authHelper, requestTracker);
```

Comment:
I'm wondering whether we want to make this mandatory or make it configurable with a switch that turns the feature on and off. I'm saying because if we need the ability of turning it off, then we need a constructor that takes no request tracker and makes it null to indicate that there is no tracking for this running instance.

Reply:
I added the possibility to disable the request tracking from configuration. Instead of overloading constructors, I added a configuration flag that is passed to the `RequestTracker`. If tracing is disabled, the cache will not even be instantiated and default request ids values will be returned by `RequestTracker`. In turn, the new logging methods check if the request id is the default one.